app/src/main/java/com/mapbox/maps/testapp/examples/CircleLayerClusteringActivity.kt

```diff
@@ -140,47 +141,31 @@ class CircleLayerClusteringActivity : AppCompatActivity() {
       intArrayOf(0, ContextCompat.getColor(this, R.color.blue))
     )
 
-    for (i in layers.indices) {
-
-      // Add clusters' circles
-      style.addLayer(
-        circleLayer("cluster-$i", GEOJSON_SOURCE_ID) {
-          circleColor(layers[i][1])
-          circleRadius(18.0)
-          filter(
-            if (i == 0) {
-              has {
-                literal("point_count")
-              }
-              gte {
-                toNumber {
-                  get { literal("point_count") }
-                }
-                literal(layers[i][0].toLong())
-              }
-            } else {
-              all {
-                has {
-                  literal("point_count")
-                }
-                gte {
-                  toNumber {
-                    get { literal("point_count") }
-                  }
-                  literal(layers[i][0].toLong())
-                }
-                lt {
-                  toNumber {
-                    get { literal("point_count") }
-                  }
-                  literal(layers[i - 1][0].toLong())
-                }
-              }
+    // Add clusters' circles
+    style.addLayer(
+      circleLayer("clusters", GEOJSON_SOURCE_ID) {
+        circleColor(
+          step{
+            get {literal("point_count")}
+            literal ( ColorUtils.colorToRgbaString(layers[2][1]) )
+            stop {
+              literal(layers[1][0].toDouble())
+              literal ( ColorUtils.colorToRgbaString(layers[1][1]) )
             }
-          )
-        }
-      )
-    }
+            stop {
+              literal(layers[0][0].toDouble())
+              literal ( ColorUtils.colorToRgbaString(layers[0][1]) )
+            }
+          }
+        )
+        circleRadius(18.0)
+        filter(
+          has {
+            literal("point_count")
+          }
```

Comment:
Nit: can be simplified to 
```suggestion
          has("point_count")
```

Reply:
@pengdev 

Thanks, fixed